src/gateway.c

```diff
@@ -700,8 +725,37 @@ static void handle_query_work_cb(struct exec *exec)
 		return;
 	}
 
-	leader_exec_result(exec, rc == SQLITE_DONE ? RAFT_OK : RAFT_ERROR);
-	TAIL return leader_exec_resume(exec);
+	leader_exec_result(exec, rc == SQLITE_DONE ? RAFT_OK :RAFT_ERROR);		
+	return leader_exec_resume(exec);
+}
+
+static void handle_query_work_cb(struct exec *exec)
+{
+	struct gateway *g = exec->data;
+	PRE(g->req != NULL && g->leader != NULL);
+	struct handle *req = g->req;
+
+	if (req->cancellation_requested) {
+		/* Nothing else to do. */
+		TAIL return leader_exec_resume(exec);
+	}
+
+	if (!sqlite3_statement_empty(exec->leader->conn, exec->tail)) {
```

Comment:
It is not a comment to the current line.
`sqlite3_statement_empty()` pretends to be a part of `sqlite3` still it's defined by us. Is it possible to rename it so that nobody has an impression that it's `sqlite3_` namespace.